synapseutils/copy.py

```diff
@@ -151,32 +188,27 @@ def _copyFile(syn, entity, destinationId, version=None, replace=False, setProven
     elif setProvenance == "existing":
         try:
             act = syn.getProvenance(ent.id)
-        except Exception as e:
-            if e.message.find('No activity') >= 0:
+        except SynapseHTTPError as e:
+            if e.response.code == 404:
                 act = None
             else:
                 raise e
     elif setProvenance is None or setProvenance.lower() == 'none':
         act = None
     else:
         raise ValueError('setProvenance must be one of None, existing, or traceback')
-    #Grab file handle createdBy annotation to see the user that created fileHandle
-    fileHandleList = syn.restGET('/entity/%s/version/%s/filehandles'%(ent.id,ent.versionNumber))
+    #Grab entity bundle
+    bundle = syn._getEntityBundle(ent.id, version=ent.versionNumber)
     #NOTE: May not always be the first index (need to filter to make sure not PreviewFileHandle)
-    #Loop through to check which dataFileHandles match and return createdBy
-    for fileHandle in fileHandleList['list']:
-        if fileHandle['id'] == ent.dataFileHandleId:
-            createdBy = fileHandle['createdBy']
-            break
-    else:
-        createdBy = None
+    fileHandle = synapseclient.utils.find_data_file_handle(bundle)
```

Comment:
getting entityBundle is wastefull.
utils.find_data_file_handle({fileHandles: filehandleList})